### PR TITLE
Fix portal route matching

### LINKED ISSUE #5489: File name "search" activates WebsiteSearchController
| Q | A
| --- | ---
| Sulu Version | 2.1.2

# Actual Behavior

```
Sulu\Component\Rest\Exception\MissingParameterException:
Missing parameter q in Sulu\Bundle\SearchBundle\Controller\WebsiteSearchController

  at vendor/sulu/sulu/src/Sulu/Component/Rest/RequestParametersTrait.php:38
  at Sulu\Bundle\SearchBundle\Controller\WebsiteSearchController->getRequestParameter(object(Request), 'q', true)
     (vendor/sulu/sulu/src/Sulu/Bundle/SearchBundle/Controller/WebsiteSearchController.php:69)
  at Sulu\Bundle\SearchBundle\Controller\WebsiteSearchController->queryAction(object(Request))
     (vendor/symfony/http-kernel/HttpKernel.php:158)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:80)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:201)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:63)
```

# Expected Behavior

show the image that was requested via ```/build/website/icons/search.svg```

# Steps to Reproduce

In a Sulu System call any file in any subfolder named "search" (+ any extension or none at all)

# Possible Solutions

No concrete ideas, but the WebsiteSearchController should not respond every time a file has the name "search". Workaround for me was to suffix all icons with '-icon'

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes #5489
| Related issues/PRs | -
| License | MIT
| Documentation PR | -

#### What's in this PR?

Fix portal route matching.

#### Why?

Currently routes e.g.: `/_profiler/search`  does also match our search, only routes with a specific portal prefix should match.

#### To Do

- [ ] Add Tests

### CODE REVIEW

**Comment by nnatter** [comment]
A functional test that tests if the generated routes match the correct paths would be nice 🙂

File: src/Sulu/Bundle/WebsiteBundle/Routing/PortalLoader.php [hunk 1]
```diff
@@ -22,32 +23,41 @@
...
         foreach ($importedRoutes as $importedRouteName => $importedRoute) {
-            $importedCondition = $importedRoute->getCondition();
-
             $collection->add(
                 $importedRouteName,
                 new PortalRoute(
                     '{prefix}' . $importedRoute->getPath(),
                     $importedRoute->getDefaults(),
-                    \array_merge(['prefix' => '.*', 'host' => '.+'], $importedRoute->getRequirements()),
+                    \array_merge(['prefix' => implode('|', $prefixes)], $importedRoute->getRequirements()),
```
Comment (nnatter) [inline src/Sulu/Bundle/WebsiteBundle/Routing/PortalLoader.php:55]: this leads to an error when the result of `implode('|', $prefixes)` is an empty string 😕

Comment (alexander-schranz) [reply src/Sulu/Bundle/WebsiteBundle/Routing/PortalLoader.php:55]: what error do you get? empty string would be the correct value then.

File: src/Sulu/Bundle/WebsiteBundle/Routing/PortalLoader.php [hunk 2]
```diff
@@ -22,32 +23,46 @@
...
+        foreach ($this->webspaceManager->getPortalInformations() as $portalInformation) {
+            if (!$portalInformation->getPrefix()) {
+                $prefixes[] = '(^$)?';
+                
+                continue;
+            }
 
-        foreach ($importedRoutes as $importedRouteName => $importedRoute) {
-            $importedCondition = $importedRoute->getCondition();
+            $prefixes[] = preg_quote($portalInformation->getPrefix());
```
Comment (nnatter) [inline src/Sulu/Bundle/WebsiteBundle/Routing/PortalLoader.php:51]: think we can just use 
```
            // symfony does not accept an empty regex as requirement, therefore we use '()' to match an empty prefix
            $prefixes[] = \preg_quote($portalInformation->getPrefix()) ?: '()';
```

**Comment by nnatter** [comment]
Would you mind to backport the test cases from https://github.com/sulu/sulu/pull/5498/files into this PR?

**Comment by alexander-schranz** [comment]
close in favor of #5498

**Pull request closed without merging**